dvc/output.py

```diff
@@ -890,9 +894,19 @@ def checkout(
         checkpoint_reset: bool = False,
         **kwargs,
     ) -> Optional[Tuple[bool, Optional[bool]]]:
+        # callback passed act as a aggregate callback.
+        # do not let checkout to call set_size and change progressbar.
+        class CallbackProxy(Callback):
```

Comment:
I don't like this that it passes an aggregated callback, but this seemed easier to do for now.